File: 2_course/prog_langs/lab1/purchase.c
#include "purchase.h"
#include <stdio.h>
#include <string.h>

void purchase_init(Purchase* p, const char* name, const char* comment, double amount, int day, int month, int year) {
    strncpy_s(p->name, sizeof(p->name), name, 63); p->name[63] = 0;
    strncpy_s(p->comment, sizeof(p->comment), comment, 127); p->comment[127] = 0;
    p->amount = amount;
    p->day = day; p->month = month; p->year = year;
}

void purchase_print(const Purchase* p, int index) {
    printf("%d. %s | %s | %.2f | %02d.%02d.%04d\n", index, p->name, p->comment, p->amount, p->day, p->month, p->year);
}

File: 2_course/prog_langs/lab2/Purchase.cs
namespace lab2;

public class Purchase
{
    public string Name { get; set; } = string.Empty;
    public string Comment { get; set; } = string.Empty;
    public double Amount { get; set; }
    public int Day { get; set; }
    public int Month { get; set; }
    public int Year { get; set; }

    public Purchase() { }

    public Purchase(string name, string comment, double amount, int day, int month, int year)
    {
        Name = name;
        Comment = comment;
        Amount = amount;
        Day = day;
        Month = month;
        Year = year;
    }

    public DateTime GetDate()
    {
        return new DateTime(Year, Month, Day);
    }

    public string ToFileString()
    {
        return $"{Name}|{Comment}|{Amount}|{Day}|{Month}|{Year}";
    }

    public static Purchase FromFileString(string line)
    {
        var parts = line.Split('|');
        if (parts.Length != 6)
            throw new FormatException("invalid file format");

        return new Purchase(
            parts[0],
            parts[1],
            double.Parse(parts[2]),
            int.Parse(parts[3]),
            int.Parse(parts[4]),
            int.Parse(parts[5])
        );
    }

    public void Print(int index)
    {
        Console.WriteLine($"{index}. {Name} | {Comment} | {Amount:F2} | {Day:D2}.{Month:D2}.{Year:D4}");
    }
}
